plugins/android_boot/fu-android-boot-device.c

```diff
@@ -0,0 +1,441 @@
+/*
+ * Copyright (C) 2022 Dylan Van Assche <me@dylanvanassche.be>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+#include <fcntl.h>
+#include <string.h>
+
+#include "fu-android-boot-device.h"
+
+#define PROC_CMDLINE	 "/proc/cmdline"
+#define MAX_CMDLINE_SIZE 4096
+#define BOOT_SLOT_ARG	 "androidboot.slot_suffix"
+#define ABL_VERSION_ARG	 "androidboot.abl.revision"
+#define SERIAL_ARG	 "androidboot.serialno"
+#define UNKNOWN_VERSION	 "0.0.UNKNOWN"
+#define SECTOR_SIZE	 512
+
+struct _FuAndroidBootDevice {
+	FuUdevDevice parent_instance;
+	const char *label;
+	const char *boot_slot;
+	const char *uuid;
+	const char *version;
+	guint64 max_size;
+};
+
+G_DEFINE_TYPE(FuAndroidBootDevice, fu_android_boot_device, FU_TYPE_UDEV_DEVICE)
+
+static char *
+fu_android_boot_extract_cmdline_arg_value(const char *key)
```

Comment:
src/fu-engine.c parses cmdline too, maybe you can copy that?

Reply:
Or preferably - make a helper in libfwupdplugin.

Reply:
Will have a look to adapt it to such a thing.